construct/core.py

```diff
@@ -1398,7 +1398,9 @@ def _build(self, obj, stream, context, path):
                 raise UnionError("none of subcons %s were found in the dictionary %s" % (self.subcons, obj))
         if isinstance(self.buildfrom, int):
             sc = self.subcons[self.buildfrom]
-            return sc.subcon._build(obj[sc.name], stream, context, path)
+            # pass the full object if building an embedded union member
+            data = obj if sc.flagembedded else obj[sc.name]
```

Comment:
No, obj2 is a name also used throughout the entire code. I tell you this because it already matches the rest of the code. Make it so. (Trek quote... just for fun. :)